implementations/micrometer-registry-prometheus/src/main/java/io/micrometer/prometheusmetrics/PrometheusMeterRegistry.java

```diff
@@ -85,17 +93,17 @@ public PrometheusMeterRegistry(PrometheusConfig config, PrometheusRegistry regis
      * @param config configuration
      * @param registry prometheus registry
      * @param clock clock
-     * @param exemplarSampler exemplar sampler
+     * @param exemplarSamplerFactory exemplar sampler factory
      */
     public PrometheusMeterRegistry(PrometheusConfig config, PrometheusRegistry registry, Clock clock,
-            @Nullable ExemplarSampler exemplarSampler) {
+            @Nullable ExemplarSamplerFactory exemplarSamplerFactory) {
```

Comment:
The advantage of providing an `ExemplarSamplerFactory` is that users can control sampling behavior, i.e.: tweak sampling config or completely redefine its logic.
We provided this possibility with the 0.x client (since we let them inject `ExemplarSampler`) and I remember users overriding it to remove exemplars on `_count`. 

Maybe it can also help testing since right now you need to wait 90ms between recordings, otherwise no Exemplar will be sampler, with a custom sampler or tweaking with sampling config, this can be avoided.

I'm ok replacing it to `SpanContext` since it is simpler and if users will ask for it we can add an extra ctor.

Reply:
I think if we make sure the Prometheus client can be configured via its standard way (see other comment thread), users will be able to control exemplar sampling interval and exemplars on count etc without us exposing a separate interface or accepting a properties object, or am I still missing something?

If I'm not missing anything, I think we can make these new classes not public to limit new API we need to maintain. It'd be great to get feedback from users on if the API we're exposing makes sense for their needs or not. Unfortunately we won't have much time since this won't make it in until the RC.